sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -325,180 +355,130 @@ class SessionCatalog(
     new Path(new Path(dbLocation), formatTableName(tableIdent.table)).toString
   }
 
-  // -------------------------------------------------------------
-  // | Methods that interact with temporary and metastore tables |
-  // -------------------------------------------------------------
+  // ----------------------------------------------
+  // | Methods that interact with temporary views |
+  // ----------------------------------------------
 
   /**
-   * Create a temporary table.
+   * Create a temporary view.
    */
   def createTempView(
       name: String,
-      tableDefinition: LogicalPlan,
-      overrideIfExists: Boolean): Unit = synchronized {
```

Comment:
If we change it back to `HashMap`, we need to add `synchronized` back. Is my understanding right?